feat: file skeleton necessary for icicle chart visualization

cguan7 opened the pull request:
Implementing a package for Icicle Chart visualizations. In order to have smaller PRs and merge small chunks of code, this initial PR will be to lay out the foundation of the files to get started!

@khtruong @mrmcduff @kristw @williaster

cguan7 committed:
feat: file skeleton necessary for icicle chart visualization

mrmcduff commented on packages/superset-ui-plugin-chart-icicle/src/Icicle.tsx:
```diff
@@ -0,0 +1,46 @@
+};
+
+export default class IcicleChart extends PureComponent<Props> {
+  private chartRef = React.createRef<HTMLDivElement>();
```
Worth considering now (since this is the skeleton) -- should this be a functional component, particularly now that giving those lightweight state with hooks is easy? My default functional/not-functional determination is "does this element need direct access to the eventual reducer?"

I haven't thought much about whether or not D3 means classes are way better or not. In any case, it's a judgment call; just wanted to point it out.

cguan7 replied:
Good point, I haven't given functional components too much thought in relation to the Icicle Chart and D3. I will look into it and see if it's worth going down that route instead.

mrmcduff commented on packages/superset-ui-plugin-chart-icicle/types/external.d.ts:
```diff
@@ -0,0 +1,4 @@
+declare module '*.png' {
```
What's this going to be used for? I've never actually seen a declaration of a filetype module.
(Not a change request -- just wondering what the pattern was)

cguan7 replied:
It's a pattern that I saw the other superset-ui-plugins follow --- it's used to create a chart's metadata which includes a thumbnail image of what an example plugin of this type would look like

mrmcduff commented on packages/superset-ui-plugin-chart-icicle/src/IciclePath.tsx:
```diff
@@ -0,0 +1,31 @@
+  setFocusById: (id: string) => console.log('[setFocusById] id: ', id),
+};
+
+export default class IciclePath extends PureComponent<Props> {
```
Same question here about functional components. Up to you.

cguan7 committed:
feat: name changes and restructure components

update package to account for clearer name and reorganized implementation for planned features

cguan7 committed:
feat: fix lint errors

cguan7 committed:
docs: add license file headers, adjust naming and links

Add Apache license file headers to all files, adjust naming of the viz in the package.json and
update the README to inform that Storybook link is unavailable but will be in the near future
